### PR TITLE
Improve error message for conflicting priorities

### PR DESCRIPTION
**Pull request by teto**
I find the error message 'nix-env --set-flag priority NUMBER PKGNAME'
not as helpful as it could be :
- doesn't share the current priorities
- doesn't say that the command must be run on the already installed
PKGNAME (which is confusing the first time)
- the doc needs careful reading:
"If there are multiple derivations matching a name in args that have the same name (e.g., gcc-3.3.6 and gcc-4.1.1), then the derivation with the highest priority is used."
if one stops reading there, he is screwed. Salvation comes with reading "A derivation can define a priority by declaring the meta.priority attribute. This attribute should be a number, with a higher value denoting a lower priority. The default priority is 0."
To sum it up, lower number wins. I tried to convey this idea in the
message too.

I haven't used this in production because nixUnstable tends to break my install and I wanted to check for agreement first.

### CODE REVIEW

**Comment by domenkozar** [comment]
Thank you!

**Comment by teto** [comment]
Thanks for the merge. The previous message did confuse me at the beginning (I know it's true for others too) and I hope this one might help beginners.
